### PR TITLE
Add DropDownList keypress while list is dropped.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This PR partially addresses issue #969 "Focus dropdown list box gets clipped when it doesn't fit in the window", by refactoring DropDownList to allow key presses to work while the drop down list is open.  It was the easiest of the suggestions that @Vezzra made to make work.

It refactors behavior that was already present when the drop down was closed, and moves it into the drop down so that it can also run when the drop down is both closed and open.

### CODE REVIEW

File: GG/src/DropDownList.cpp [hunk 1]
```diff
@@ -53,13 +58,40 @@ class ModalListPicker : public Wnd
 
     mutable SelChangedSignalType SelChangedSignal; ///< the selection change signal object for this DropDownList
 
+    DropDownList::iterator CurrentItem();
+
+    /** Select \p it  in m_lb_wnd is not none and return the new selected or none if no change.*/
```
Comment (dbenage-cx) [inline GG/src/DropDownList.cpp]: This comment is hard to read, can it be clarified?

**Review by dbenage-cx — APPROVED** [review]
No issue with selecting using keys or mouse wheel here.
Not sure this PR is a complete fix for the issue, since the list itself is still 'clipped'..

File: GG/src/DropDownList.cpp (hunk 1 above)
Comment (LGM-Doyle) [reply GG/src/DropDownList.cpp]: It looks like I left out the word "if" as well as phrased is awkwardly.  I've rephrased it.

File: GG/src/DropDownList.cpp [hunk 2]
```diff
@@ -53,13 +58,41 @@ class ModalListPicker : public Wnd
...
+    virtual void    MouseWheel(const Pt& pt, int move, Flags<ModKey> mod_keys);
+
 private:
     void LBSelChangedSlot(const ListBox::SelectionSet& rows);
 
     void LBLeftClickSlot(ListBox::iterator it, const GG::Pt& pt, const GG::Flags<GG::ModKey>& modkeys);
 
     ListBox*    m_lb_wnd;
     const Wnd*  m_relative_to_wnd;
+    bool m_dropped;  ///< Is the drop down list open.
```
Comment (geoffthemedio) [inline GG/src/DropDownList.cpp:95]: indent to align with other members

**Comment by Vezzra** [comment]
Thanks! 👍

**Pull request merged**